pkg/networkservice/core/eventchannel/monitor_client.go

```diff
@@ -50,13 +50,16 @@ func NewMonitorConnectionClient(eventCh <-chan *networkservice.ConnectionEvent)
 	return rv
 }
 
-func (m *monitorConnectionClient) MonitorConnections(ctx context.Context, in *networkservice.MonitorScopeSelector, _ ...grpc.CallOption) (networkservice.MonitorConnection_MonitorConnectionsClient, error) {
+func (m *monitorConnectionClient) MonitorConnections(ctx context.Context, _ *networkservice.MonitorScopeSelector, _ ...grpc.CallOption) (networkservice.MonitorConnection_MonitorConnectionsClient, error) {
 	fanoutEventCh := make(chan *networkservice.ConnectionEvent, 100)
 	m.updateExecutor.AsyncExec(func() {
 		m.fanoutEventChs = append(m.fanoutEventChs, fanoutEventCh)
 		go func() {
 			<-ctx.Done()
 			m.updateExecutor.AsyncExec(func() {
+				if len(m.fanoutEventChs) == 0 {
+					return
+				}
```

Comment:
Can this check be moved outside the lambda, right after <-ctx.Done()?

Reply:
I think no, because we need synchronized access to `m.fanoutEventChs`, that guaranteed by `m.updateExecutor`